packages/backend/src/open_payments/quote/service.ts

```diff
@@ -116,69 +106,52 @@ async function createQuote(
 
   try {
     const receiver = await resolveReceiver(deps, options)
-    const ilpQuote = await startQuote(deps, {
-      ...options,
+    const quote = await deps.paymentMethodHandlerService.getQuote('ILP', {
       paymentPointer,
-      receiver
+      receiver,
+      receiveAmount: options.receiveAmount,
+      debitAmount: options.debitAmount
     })
 
+    const maxPacketAmount = quote.additionalFields.maxPacketAmount as bigint
+
     const sendingFee = await deps.feeService.getLatestFee(
       paymentPointer.assetId,
       FeeType.Sending
     )
 
     return await Quote.transaction(deps.knex, async (trx) => {
-      const quote = await Quote.query(trx)
+      const createdQuote = await Quote.query(trx)
         .insertAndFetch({
           paymentPointerId: options.paymentPointerId,
           assetId: paymentPointer.assetId,
           receiver: options.receiver,
-          debitAmount: {
-            value: ilpQuote.maxSourceAmount,
-            assetCode: paymentPointer.asset.code,
-            assetScale: paymentPointer.asset.scale
-          },
-          receiveAmount: {
-            value: ilpQuote.minDeliveryAmount,
-            assetCode: receiver.assetCode,
-            assetScale: receiver.assetScale
-          },
+          debitAmount: quote.debitAmount,
+          receiveAmount: quote.receiveAmount,
           // Cap at MAX_INT64 because of postgres type limits.
           maxPacketAmount:
```

Comment:
Thoughts on refactoring Quote so that ILP-related fields are more delineated from "base" quote stuff? I was thinking it might help keep the amount of columns in quote reasonable when other payment methods become supported.

Reply:
💯 
I have it in a separate issue:
- https://github.com/interledger/rafiki/issues/2000